packages/hash/frontend/src/shared/readonly-mode/context.tsx

```diff
@@ -0,0 +1,37 @@
+import { useRouter } from "next/router";
+import { createContext, ReactNode, useContext, useMemo } from "react";
+
+type ReadonlyModeInfo = {
+  readonlyMode: boolean;
+};
+
+const ReadonlyModeContext = createContext<ReadonlyModeInfo>({
+  readonlyMode: false,
+});
+
+export const ReadonlyModeProvider = ({
+  children,
+}: {
+  children?: ReactNode;
+}) => {
+  const router = useRouter();
+
+  const contextValue = useMemo(
+    () => ({
+      readonlyMode: "readonly" in router.query,
```

Comment:
We know that this is a temporary solution but this is not obvious from the codebase itself. If someone opens this file, they may get confused. I’d place a comment to communicate our intent to others and future selves. Perhaps, the right place for this is a JSDoc for `const ReadonlyModeProvider`

Reply:
I agree. Updated in [a74f0d5](https://github.com/hashintel/hash/pull/893/commits/a74f0d55f7c9b7806d3b5ab86f07c41f1644fe6a)